modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2168,12 +2168,29 @@ def map(
         PandasDataframe
             A new dataframe.
         """
-        map_fn = (
-            self._partition_mgr_cls.lazy_map_partitions
-            if lazy
-            else self._partition_mgr_cls.map_partitions
-        )
-        new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        if self.num_parts <= CpuCount.get():
+            map_fn = (
+                self._partition_mgr_cls.lazy_map_partitions
+                if lazy
+                else self._partition_mgr_cls.map_partitions
+            )
+            new_partitions = map_fn(self._partitions, func, func_args, func_kwargs)
+        else:
+            axis = (
+                0
+                if abs(self._partitions.shape[0] - CpuCount.get())
+                < abs(self._partitions.shape[1] - CpuCount.get())
+                else 1
```

Comment:
```suggestion
                1
                if abs(self._partitions.shape[0] - CpuCount.get())
                < abs(self._partitions.shape[1] - CpuCount.get())
                else 0
```
0 means column parts and 1 means row parts. Shouldn't these be swapped?